packages/runtime-core/src/renderer.ts

```diff
@@ -2177,7 +2177,8 @@ function baseCreateRenderer(
         (type === Fragment &&
           patchFlag &
             (PatchFlags.KEYED_FRAGMENT | PatchFlags.UNKEYED_FRAGMENT)) ||
-        (!optimized && shapeFlag & ShapeFlags.ARRAY_CHILDREN)
+        ((!optimized || patchFlag === PatchFlags.BAIL) &&
```

Comment:
I dont think this is the right fix. the optimized should set to false if patchFlag equals BAIL.

Reply:
Thanks for your review. I've updated the code.